pdfly/rotate.py

```diff
@@ -0,0 +1,81 @@
+"""
+Rotate specified pages by the specified amount
+
+Example:
+    pdfly rotate --output output.pdf input.pdf 90
+        Rotate all pages by 90 degrees (clockwise)
+
+A file not followed by a page range (PGRGS) means all the pages of the file.
+
+PAGE RANGES are like Python slices.
+
+        Remember, page indices start with zero.
+
+        Page range expression examples:
+
+            :     all pages.                   -1    last page.
+            22    just the 23rd page.          :-1   all but the last page.
+            0:3   the first three pages.       -2    second-to-last page.
+            :3    the first three pages.       -2:   last two pages.
+            5:    from the sixth page onward.  -3:-1 third & second to last.
+
+        The third, "stride" or "step" number is also recognized.
+
+            ::2       0 2 4 ... to the end.    3:0:-1    3 2 1 but not 0.
+            1:10:2    1 3 5 7 9                2::-1     2 1 0.
+            ::-1      all pages in reverse order.
+
+
+"""
+
+# Copyright (c) 2014, Steve Witham <switham_github@mac-guyver.com>.
+# All rights reserved. This software is available under a BSD license;
+# see https://github.com/py-pdf/pypdf/LICENSE
+
+import sys
+import traceback
+from pathlib import Path
+from typing import Set
+
+from pypdf import (
+    PageRange,
+    PdfReader,
+    PdfWriter,
+)
+
+
+def main(
+    filename: Path,
+    output: Path,
+    degrees: int,
+    page_range: str,
+) -> None:
+    try:
+        # Set up the streams
+        reader = PdfReader(filename)
+        pages = list(reader.pages)
+        writer = PdfWriter()
+
+        # Convert the page range into a set of page numbers
+        pages_to_rotate = convert_range_to_pages(page_range, len(pages))
+
+        for page_index, page in enumerate(pages):
+            if page_index in pages_to_rotate:
+                writer.add_page(page.rotate(degrees))
+            else:
+                writer.add_page(page)
+
+        # Everything looks good! Write the output file.
+        if True:
```

Comment:
Why `if True`?

Were you planning to handle writing to stdout there?

Reply:
Another debug leftover ☹️